Sources/Workspace/Workspace.swift

```diff
@@ -593,9 +593,13 @@ extension Workspace {
     /// - Parameters:
     ///     - diagnostics: The diagnostics engine that reports errors, warnings
     ///       and notes.
+    ///     - dryRun: Determines if changes will be made to the checkout, dependency manifests and pins store.
+    ///     - changes: Array to fill for dry run.
     public func updateDependencies(
         root: PackageGraphRootInput,
-        diagnostics: DiagnosticsEngine
+        diagnostics: DiagnosticsEngine,
+        dryRun: Bool = false,
+        changes: inout [(PackageReference, PackageStateChange)]
```

Comment:
What if make this optional with default value as nil so we:
1. Avoid passing in the dryRun bool
2. Retain API compatibility.